### PR TITLE
Bug/fixed pooling

### LINKED ISSUE #20: Shooting Component
A shooting component should be created to manage the shooting of projectiles for every entity in the game. The player and enemies should be handled differently, such that player shooting so be handled by inputs received from events, and the enemies by their AI states.

The shooting component should sit between the projectile manager as such.

```Entity``` -> ```Shooting Component``` -> ```Projectile Manager``` -> ```Projectiles```

### PR DESCRIPTION
**Pull request by BlackRece**
**Describe your changes**
yeah

**Screenshots**
If applicable, add screenshots to help explain your changes.

**Additional context**
Add any other context about your changes here.

### CODE REVIEW

**Review by kyle-robinson — COMMENTED** [review]
Looks good to me 😎 👍🏻 

Just wanted to ask about these though. Does shooting still work correctly with the player after removing the check for the first night?

File: Roche-Engine/Components/PlayerShooting.cpp [hunk 1]
```diff
@@ -26,7 +26,7 @@ void PlayerShooting::Update(float dt)
 	}
 
 	// handle mouse shooting actions/event
-	if (m_bIsShooting && shootingDelay < 0 && !firstNight)
+	if (m_bIsShooting && shootingDelay < 0 /*&& !firstNight*/)
```
Comment (kyle-robinson) [inline Roche-Engine/Components/PlayerShooting.cpp]: ^^

File: Roche-Engine/Components/PlayerShooting.cpp [hunk 2]
```diff
@@ -81,7 +81,7 @@ void PlayerShooting::HandleEvent(Event* event)
 	case EVENTID::LeftMouseRelease:
 	{
 		m_pEmitter->SetTargetPosition(m_vMousePos);
-		m_bIsShooting = false;
+		//m_bIsShooting = false;
```
Comment (kyle-robinson) [inline Roche-Engine/Components/PlayerShooting.cpp:84]: ^^

Comment (juliusz-kaczmarek) [reply Roche-Engine/Components/PlayerShooting.cpp:84]: Fixed it

**Review by juliusz-kaczmarek — APPROVED** [review]
Rece, you're a champ, looks good to me, good job 🥇 🔥

**Pull request merged**